packages/techdocs-common/src/stages/publish/azureBlobStorage.ts

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import path from 'path';
+import express from 'express';
+import {
+  BlobServiceClient,
+  BlobUploadCommonResponse,
+  StorageSharedKeyCredential,
+} from '@azure/storage-blob';
+import { Logger } from 'winston';
+import { Entity, EntityName } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { getHeadersForFileExtension, getFileTreeRecursively } from './helpers';
+import { PublisherBase, PublishRequest, TechDocsMetadata } from './types';
+import limiterFactory from 'p-limit';
+import JSON5 from 'json5';
+
+// The number of batches that may be ongoing at the same time.
+const BATCH_CONCURRENCY = 3;
+
+export class AzureBlobStoragePublish implements PublisherBase {
+  static async fromConfig(
+    config: Config,
+    logger: Logger,
+  ): Promise<PublisherBase> {
+    let containerName = '';
+    try {
+      containerName = config.getString(
+        'techdocs.publisher.azureBlobStorage.containerName',
+      );
+    } catch (error) {
+      throw new Error(
+        "Since techdocs.publisher.type is set to 'awsS3' in your app config, " +
+          'techdocs.publisher.awsS3.bucketName is required.',
+      );
+    }
+
+    // Credentials is an optional config. If missing, default AWS environment variables
+    // or AWS shared credentials file at ~/.aws/credentials will be used to authenticate
+    // https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-environment.html
+    // https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-shared.html
+    let account = '';
+    let accountKey = '';
+    account =
+      config.getOptionalString(
+        'techdocs.publisher.azureBlobStorage.credentials.account',
+      ) || '';
+    accountKey =
+      config.getOptionalString(
+        'techdocs.publisher.azureBlobStorage.credentials.accountKey',
+      ) || '';
+
+    const credential = new StorageSharedKeyCredential(account, accountKey);
+    const storageClient = new BlobServiceClient(
+      `https://${account}.blob.core.windows.net`,
```

Comment:
Umm, I'm afraid if this will work. If `account` is not defined in config, this will become
`https://.blob.core.windows.net` which will throw an error.

Looking at the examples, it looks `account` is something that we may need our user to define in the config.